### ISSUE TITLE
 Incompatible shapes error and bullet #5 DID NOT fix that

### ISSUE THREAD

**Issue by Maxxiey** [keithito/tacotron#221]
Hi there, I encountered a  Incompatible shapes: [1,1595,513] vs. [1,640,513] Error during training and the ALMIGHTY bullet #5 can not fix that. This error is raised when I calculate the L1 loss between linear_target and linear_output. I am not quite fimiliar with the whole structure of the network, so is there something I missed cause I really can't figure out which part decides the total length of the linear_output?

Any suggestion on how to fix that, why would it happen, or is there anyone else who has same problem?

Part of my training log:
(uploading strangely went worng,have to paste it)

Exiting due to exception: Incompatible shapes: [1,1545,513] vs. [1,620,513]
	 [[Node: model/optimizer/gradients/model/loss/sub_1_grad/BroadcastGradientArgs = BroadcastGradientArgs[T=DT_INT32, _device="/job:localhost/replica:0/task:0/device:CPU:0"](model/optimizer/gradients/model/loss/sub_1_grad/Shape, model/optimizer/gradients/model/loss/sub_1_grad/Shape_1)]]

Caused by op 'model/optimizer/gradients/model/loss/sub_1_grad/BroadcastGradientArgs', defined at:
  File ".\train.py", line 202, in <module>
    main()
  File ".\train.py", line 198, in main
    train(log_dir, args)
  File ".\train.py", line 74, in train
    model.add_optimizer(global_step)
  File "D:\projects\tacotron\models\tacotron.py", line 174, in add_optimizer
    gradients, variables = zip(*optimizer.compute_gradients(self.loss))
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\training\optimizer.py", line 514, in compute_gradients
    colocate_gradients_with_ops=colocate_gradients_with_ops)
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\ops\gradients_impl.py", line 596, in gradients
    gate_gradients, aggregation_method, stop_gradients)
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\ops\gradients_impl.py", line 779, in _GradientsHelper
    lambda: grad_fn(op, *out_grads))
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\ops\gradients_impl.py", line 398, in _MaybeCompile
    return grad_fn()  # Exit early
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\ops\gradients_impl.py", line 779, in <lambda>
    lambda: grad_fn(op, *out_grads))
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\ops\math_grad.py", line 889, in _SubGrad
    rx, ry = gen_array_ops.broadcast_gradient_args(sx, sy)
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\ops\gen_array_ops.py", line 812, in broadcast_gradient_args
    "BroadcastGradientArgs", s0=s0, s1=s1, name=name)
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\util\deprecation.py", line 454, in new_func
    return func(*args, **kwargs)
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 3155, in create_op
    op_def=op_def)
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 1717, in __init__
    self._traceback = tf_stack.extract_stack()

...which was originally created as op 'model/loss/sub_1', defined at:
  File ".\train.py", line 202, in <module>
    main()
[elided 0 identical lines from previous traceback]
  File ".\train.py", line 198, in main
    train(log_dir, args)
  File ".\train.py", line 73, in train
    model.add_loss()
  File "D:\projects\tacotron\models\tacotron.py", line 149, in add_loss
    l1 = tf.abs(self.linear_targets - self.linear_outputs)
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\ops\math_ops.py", line 850, in binary_op_wrapper
    return func(x, y, name=name)
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\ops\gen_math_ops.py", line 8769, in sub
    "Sub", x=x, y=y, name=name)
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\util\deprecation.py", line 454, in new_func
    return func(*args, **kwargs)
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 3155, in create_op
    op_def=op_def)
  File "C:\Users\max\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 1717, in __init__
    self._traceback = tf_stack.extract_stack()

**Comment by Maxxiey** [comment]
I have fixed this problem, close this issue now.
